pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -136,29 +124,23 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 
 	groupMap := make(map[string]*apimodels.RuleGroup)
 
-	for _, r := range ruleGroupQuery.Result {
-		if len(r) < 3 {
-			continue
-		}
-		groupId, namespaceUID, namespace := r[0], r[1], r[2]
-
-		newGroup := &apimodels.RuleGroup{
-			Name:           groupId,
-			File:           namespace, // file is what Prometheus uses for provisioning, we replace it with namespace.
-			LastEvaluation: time.Time{},
-			EvaluationTime: 0, // TODO: see if we are able to pass this along with evaluation results
-		}
-
-		groupMap[groupId+"-"+namespaceUID] = newGroup
-
-		ruleResponse.Data.RuleGroups = append(ruleResponse.Data.RuleGroups, newGroup)
-	}
-
 	for _, rule := range alertRuleQuery.Result {
-		newGroup, ok := groupMap[rule.RuleGroup+"-"+rule.NamespaceUID]
+		groupKey := rule.RuleGroup + "-" + rule.NamespaceUID
+		newGroup, ok := groupMap[groupKey]
```

Comment:
If I'm reading this right, this is only a new group if `!ok`, and it's immediately created right after. So in many cases, the group is not new. `newRule` is always a newly-allocated struct, but putting `new` in the variable name reads to me like the code will take an action to create a new alerting rule somewhere.

What do you think about this variable renaming:
```
rule => resultRule
newGroup => ruleGroup (or group)
newRule => rule
```

or perhaps:
```
newRule => apiRule
alertingRule => apiAlertingRule
```

Reply:
Actually, it is opposite. The group is almost always new. Grafana UI as well as alerting scheduler do not support alert grouping.